packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -139,6 +154,11 @@ export class HttpRoute extends Resource implements IHttpRoute {
       apiId: props.httpApi.httpApiId,
       routeKey: props.routeKey.key,
       target: `integrations/${integration.integrationId}`,
+      ...props.authorizer ? {
+        authorizerId: props.authorizer.authorizerId,
+        authorizationType: props.authorizer.authorizerType,
+        authorizationScopes: props.authorizationScopes,
+      }: {},
```

Comment:
As mentioned, this needs to be implemented via `bind()`

It will look like this - 

```ts
if (props.authorizer) {
  const result = props.authorizer.bind(this);
}

const routeProps = {
  ...
  authorizerId: result.authorizerId;
  authorizationType: result.authorizationType,
  authorizationScopes: result.authorizationScopes,
};
```

Reply:
Okay cool, i guess except `authorizationScopes` which is separate from the authorizer, and are attached to the route (not apart of `AWS::ApiGatewayV2::Authorizer`).

Still not sure the reason why bind is needed but I will try to implement it this way. See  https://github.com/aws/aws-cdk/pull/10972#discussion_r519258536 first though before we come to this.

Reply:
I've responded in this comment - https://github.com/aws/aws-cdk/pull/10972#discussion_r520455862